clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -121,10 +122,14 @@ public void readInto(ByteBuffer buffer, int position) throws IOException {
     }
 
     @Override
-    public Records slice(int position, int size) throws IOException {
+    public FileRecords slice(int position, int size) {
         int availableBytes = availableBytes(position, size);
         int startPosition = this.start + position;
-        return new FileRecords(file, channel, startPosition, startPosition + availableBytes, true);
+        try {
+            return new FileRecords(file, channel, startPosition, startPosition + availableBytes, true);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
```

Comment:
In RecordBatchIterator, we convert an IOException to a KafkaException. Should we be consistent here?

Reply:
![image](https://github.com/user-attachments/assets/3ae0c29b-8d51-4cae-aae0-c4bb8b7efff2)

It seems the IOException is meaningless when the isSlice=true. Maybe we can have a new constructor of FileRecords which does not have IOException in the signature?

Reply:
> In RecordBatchIterator, we convert an IOException to a KafkaException. Should we be consistent here?

Are you suggesting I change it to throw an `UncheckedIOException`? If so, that is what I did. I looked at the method declaration and we don't seem to document that `KafkaException` is thrown. It should be okay to change the type. What do you think?

Reply:
> It seems the IOException is meaningless when the isSlice=true. Maybe we can have a new constructor of FileRecords which does not have IOException in the signature?

Good catch. Fixed it. Let me know what you think of my solution.